sdk/core/azure-core-test/src/main/java/com/azure/core/test/annotation/TestingServiceVersions.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.annotation;
+
+import com.azure.core.test.TestMode;
+import com.azure.core.util.ServiceVersion;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation provides test classes the ability to configure which service versions are used when running tests in
+ * {@link TestMode#RECORD}, {@link TestMode#LIVE}, and {@link TestMode#PLAYBACK} modes.
+ * <p>
+ * The service versions configured here mat interact with how {@link AzureMethodSource} generates parameterized test
+ * permutations.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE})
+@Inherited
+public @interface TestingServiceVersions {
+    /**
+     * The class that represents the service version type.
+     * <p>
+     * This is used to convert {@link #minimumLiveServiceVersion()}, {@link #maximumLiveServiceVersion()}, and {@link
+     * #recordingServiceVersion()} into their {@link ServiceVersion} type.
+     * <p>
+     * If neither of {@link #minimumLiveServiceVersion()} and {@link #maximumLiveServiceVersion()} are provided all
+     * {@link ServiceVersion ServiceVersions} represented by this type will be used. If {@link
+     * #recordingServiceVersion()} isn't configured the latest {@link ServiceVersion} represented by this type will be
+     * used.
+     * <p>
+     * If this type isn't an {@link Enum} an {@link IllegalArgumentException} will be thrown during runtime.
+     *
+     * @return The class that represents the service version type.
+     */
+    Class<? extends ServiceVersion> serviceVersionType();
+
+    /**
+     * The minimum service version used during {@link TestMode#LIVE} testing.
+     * <p>
+     * If neither this or {@link #maximumLiveServiceVersion()} are set all service versions will be used. If only this
```

Comment:
If both min and max versions are not set, should we just run using the latest version instead of running against all versions?

Reply:
I keep ping-ponging back and forth on this. I think you're right that by default this should only run against latest but have the ability to use an environment configuration to run against all service versions that should be supported by the test. AKA, have the ability to run a weekly regression test to ensure the SDK didn't break back-compat with a previous service version that should be supported.

Reply:
Yeah, having the env var to configure more service versions is better. By default, just using the latest version reduces the effort to configure this annotation.

Reply:
For now this concept will be removed (class-level annotation of supported service versions) as there isn't a realized scenario where this will be used at this time.